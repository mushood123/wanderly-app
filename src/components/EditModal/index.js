import { EditModal } from './editModal';

export { EditModal };
