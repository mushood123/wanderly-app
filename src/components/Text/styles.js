import {StyleSheet} from 'react-native';
import {font} from '../../theme/fonts';

export const styles = StyleSheet.create({
  defaultText: {
    fontFamily: font.primary,
  },
});
