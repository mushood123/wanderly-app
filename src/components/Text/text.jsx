import React from 'react';
import { Text as RNText } from 'react-native';
import { styles } from './styles';

export const Text = props => (
    <RNText
        {...props}
        style={{ ...styles.defaultText, ...props.style }}
    >
        {props.children}
    </RNText>
);
