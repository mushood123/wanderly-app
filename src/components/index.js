import {FormField} from './FormField';
import {Text} from './Text';

export {FormField, Text};
