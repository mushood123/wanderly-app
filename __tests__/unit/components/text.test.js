import React from 'react';
import { render } from '../utils/theme-wrapper/theme-wrapper';
import { Text } from '~src/components/Text';


describe('Text component', () => {
    it('renders correctly with default styles and matches snapshot', () => {
        const tree = render(<Text>Hello World</Text>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with custom styles and matches snapshot', () => {
        const customStyle = { color: 'red', fontSize: 20 };
        const tree = render(<Text style={customStyle}>Styled Text</Text>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with nested elements and matches snapshot', () => {
        const tree = render(
            <Text>
                <Text style={{ fontWeight: 'bold' }}>Nested Bold Text</Text>
            </Text>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
