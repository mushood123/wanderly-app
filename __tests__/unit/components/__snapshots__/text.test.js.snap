// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component renders correctly with custom styles and matches snapshot 1`] = `
<Text
  style={
    {
      "color": "red",
      "fontFamily": "Poppins-Regular",
      "fontSize": 20,
    }
  }
>
  Styled Text
</Text>
`;

exports[`Text component renders correctly with default styles and matches snapshot 1`] = `
<Text
  style={
    {
      "fontFamily": "Poppins-Regular",
    }
  }
>
  Hello World
</Text>
`;

exports[`Text component renders correctly with nested elements and matches snapshot 1`] = `
<Text
  style={
    {
      "fontFamily": "Poppins-Regular",
    }
  }
>
  <Text
    style={
      {
        "fontFamily": "Poppins-Regular",
        "fontWeight": "bold",
      }
    }
  >
    Nested Bold Text
  </Text>
</Text>
`;
